src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/Ordering/OrderingHelper.cs

```diff
@@ -165,9 +171,87 @@ public static IProjectTree GetLastChild(IProjectTree projectTree)
         {
+            Requires.NotNull(projectTree, nameof(projectTree));
+
             return GetChildren(projectTree).FirstOrDefault();
         }
 
+        /// <summary>
+        /// Get the difference to see what was added.
+        /// We do a sanity check to make sure they have a valid display order.
+        /// We also order the added nodes by their display order.
+        /// </summary>
+        public static IEnumerable<IProjectTree> GetAddedNodes(IProjectTree previousTarget, IProjectTree currentTarget)
+        {
+            Requires.NotNull(previousTarget, nameof(previousTarget));
+            Requires.NotNull(currentTarget, nameof(currentTarget));
+
+            return
+                currentTarget.Children.Where(updatedChild => 
+                {
+                    var hasChild = previousTarget.TryFind(updatedChild.Identity, out var child);
+                    var hasValidUpdatedChild = HasValidDisplayOrder(updatedChild);
+
+                    return (!hasChild || (hasChild && !HasValidDisplayOrder(child))) && hasValidUpdatedChild;
+                })
+                .OrderBy(GetDisplayOrder).ToList();
+        }
+
+        /// <summary>
+        /// This moves the nodes above the target's first child. 
+        /// If there are no children with a valid display order of the target or the target is an empty folder, it will try the parent target's first child.
+        /// </summary>
+        public static bool TryMoveNodesToTop(Project project, IEnumerable<IProjectTree> nodes, IProjectTree target)
+        {
+            Requires.NotNull(project, nameof(project));
+            Requires.NotNull(nodes, nameof(nodes));
+            Requires.NotNull(target, nameof(target));
+
+            // Move added nodes to the top. 
+            var nodeToAddAbove = GetFirstChild(target);
+
+            // This could have been an empty folder and nodeToAddAbove is null, therefore
+            //     get the first child of the target's parent.
+            // Empty folders do not have a valid display order.
+            // In the future, if we want to order empty folders, meaning they have a valid display order, 
+            //     this will need to be extended to handle that behavior.
+
+            // This recursively tries to find a node to work with. If there is none at all, this will break on project root.
+            var currentTarget = target;
+            while(nodeToAddAbove == null && currentTarget.IsFolder && !GetChildren(currentTarget).Any())
+            {
+                currentTarget = currentTarget.Parent;
+                nodeToAddAbove = GetFirstChild(currentTarget);
+            }
+
+            if (nodeToAddAbove != null && nodes.Any())
+            {
+                var didAllMoveSuccessfully = true;
+
+                foreach (var node in nodes)
+                {
+                    var didMove = TryMoveAbove(project, node, nodeToAddAbove);
```

Comment:
You could inline this method call to just be `if (!TryMoveAbove(project, node, nodeToAddAbove))`

Reply:
Yes. I actually had this and it didn't get included.

Reply:
Done